src/main/java/com/griddynamics/gridu/phonebookmvc/service/PhonebookServiceIml.java

```diff
@@ -0,0 +1,76 @@
+package com.griddynamics.gridu.phonebookmvc.service;
+
+import com.griddynamics.gridu.phonebookmvc.entity.Contact;
+import com.griddynamics.gridu.phonebookmvc.repository.ContactRepository;
+import com.griddynamics.gridu.phonebookmvc.exception.ContactAlreadyExistException;
+import com.griddynamics.gridu.phonebookmvc.exception.ContactNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class PhonebookServiceIml implements  PhonebookService {
+
+    @Autowired
+    private ContactRepository repository;
+
+    @Override
+    public Contact saveContact(Contact contact) throws ContactAlreadyExistException {
+        try {
+            getContactByName(contact.getName());
```

Comment:
Well, you could add a method like ``doesExist(long id)`` or something like that. That way you would save some number of code lines and simplify the logic a bit.

Reply:
I added method doesContactExist(Contact contact) to make logic a bit simpler and readable.